src/book/entities/Account.ts

```diff
@@ -148,69 +129,26 @@ export default class Account extends BaseEntity {
     const account = await super.save(options);
 
     if (this.queryClient) {
-      updateCache({ queryClient: this.queryClient, entity: this });
+      updateCache({ queryClient: this.queryClient });
     }
 
     return account;
   }
 
   async remove(options?: SaveOptions): Promise<this> {
     if (this.queryClient) {
-      updateCache({ queryClient: this.queryClient, entity: this, isDelete: true });
+      updateCache({ queryClient: this.queryClient });
     }
 
     const account = await super.remove(options);
     return account;
   }
 }
 
-/**
- * Because accounts have parent/children relations,
- * whenever an account is updated or deleted we need to update
- * those relations
- */
-export async function updateCache(
-  {
-    queryClient,
-    entity,
-    isDelete = false,
-  }: {
-    queryClient: QueryClient,
-    entity: Account,
-    isDelete?: boolean,
-  },
-) {
-  // We need this to reload childrenIds... haven't explored much
-  // maybe there is a better way.
-  await entity.reload();
-
-  queryClient.setQueryData(
-    [...Account.CACHE_KEY],
-    (entities: Account[] | undefined) => {
-      if (!entities) {
-        return undefined;
-      }
-
-      const newEntities = [...entities];
-
-      if (entity.type !== 'ROOT') {
-        const parentIndex = entities.findIndex(e => e.guid === entity.parentId) as number;
-        const parentAccount = entities[parentIndex];
-        const childIndex = parentAccount.childrenIds.findIndex(
-          (guid: string) => guid === entity.guid,
-        );
-        if (childIndex < 0) {
-          parentAccount.childrenIds.push(entity.guid);
-        } else if (isDelete) {
-          parentAccount.childrenIds.splice(childIndex, 1);
-        }
-
-        newEntities[parentIndex] = parentAccount;
-      }
-
-      return newEntities;
-    },
-  );
+export async function updateCache({ queryClient }: { queryClient: QueryClient }) {
```

Comment:
simplifying this for now as re-computing accounts is not expensive as there are not many. May re-optimise in the future but this removes complexity